evalml/pipelines/components/estimators/regressors/exponential_smoothing_regressor.py

```diff
@@ -0,0 +1,151 @@
+        trend=None,
+        damped_trend=False,
+        seasonal=None,
+        sp=None,
+        use_boxcox=None,
+        n_jobs=-1,
+        random_seed=0,
+        **kwargs,
+    ):
+        parameters = {
+            "trend": trend,
+            "damped_trend": damped_trend,
+            "seasonal": seasonal,
+            "sp": sp,
+            "use_boxcox": use_boxcox,
+        }
+        parameters.update(kwargs)
+        smoothing_model_msg = (
+            "sktime is not installed. Please install using `pip install sktime.`"
+        )
+        sktime_smoothing = import_or_raise(
+            "sktime.forecasting.exp_smoothing", error_msg=smoothing_model_msg
+        )
+        smoothing_model = sktime_smoothing.ExponentialSmoothing(**parameters)
+
+        super().__init__(
+            parameters=parameters,
+            component_obj=smoothing_model,
+            random_seed=random_seed,
+        )
+
+    def _remove_datetime(self, data, features=False):
+        if data is None:
+            return None
+        data_no_dt = data.copy()
+        if isinstance(
+            data_no_dt.index, (pd.DatetimeIndex, pd.PeriodIndex, pd.IntervalIndex)
+        ):
+            data_no_dt = data_no_dt.reset_index(drop=True)
+        if features:
+            data_no_dt = data_no_dt.select_dtypes(exclude=["datetime64"])
+
+        return data_no_dt
+
+    def _match_indices(self, X, y):
+        if X is not None:
+            y.index = X.index
+        return X, y
+
+    def _set_forecast(self, X):
+        from sktime.forecasting.base import ForecastingHorizon
+
+        fh_ = ForecastingHorizon([i + 1 for i in range(len(X))], is_relative=True)
+        return fh_
+
+    def fit(self, X, y=None):
+        """Fits Exponential Smoothing Regressor to data.
+
+        Args:
+            X (pd.DataFrame): The input training data of shape [n_samples, n_features].
+            y (pd.Series): The target training data of length [n_samples].
+
+        Returns:
+            self
+
+        Raises:
+            ValueError: If y was not passed in.
+        """
+        X, y = self._manage_woodwork(X, y)
+        if y is None:
+            raise ValueError("Exponential Smoothing Regressor requires y as input.")
+
+        X = self._remove_datetime(X, features=True)
+        y = self._remove_datetime(y)
+        X, y = self._match_indices(X, y)
+
+        if X is not None and not X.empty:
+            self._component_obj.fit(y=y, X=X)
+        else:
+            self._component_obj.fit(y=y)
+        return self
+
+    def predict(self, X, y=None):
+        """Make predictions using fitted Exponential Smoothing regressor.
+
+        Args:
+            X (pd.DataFrame): Data of shape [n_samples, n_features].
+            y (pd.Series): Target data.
+
+        Returns:
+            pd.Series: Predicted values.
+        """
+        X, y = self._manage_woodwork(X, y)
+        fh_ = self._set_forecast(X)
+        X = X.select_dtypes(exclude=["datetime64"])
```

Comment:
@eccabay This is looking great. I'm excited to see what the perf tests look like.

I don't think we need all these modifications to `X`, e.g. `select_dtypes`, `_match_indices`, `remove_datetime`.

Holt-Winters is a smoothing technique that relies only on the target variables `y`. You can see the source code here in sktime that ignores the `X`: https://github.com/alan-turing-institute/sktime/blob/v0.9.0/sktime/forecasting/exp_smoothing.py#L104

I think we can remove all of this logic since it's not needed for the estimator?